phoenix-core/src/main/java/org/apache/phoenix/compile/CreateTableCompiler.java

```diff
@@ -128,6 +128,24 @@ public MutationPlan compile(CreateTableStatement create) throws SQLException {
             viewTypeToBe = parentToBe.getViewType() == ViewType.MAPPED ? ViewType.MAPPED : ViewType.UPDATABLE;
             if (whereNode == null) {
                 viewStatementToBe = parentToBe.getViewStatement();
+
```

Comment:
This code is in a whereNode==null condition.  What if my view had a simple where clause in it would we allow the construction?

Reply:
didn't look at the Phoenix supported gramma carefully, I moved code outside wherenode check ifelse block.